packages/frame-core/src/types.ts

```diff
@@ -172,3 +177,17 @@ export type EmitEthProvider = <event extends EthProviderWireEvent["event"]>(
   event: event,
   params: Extract<EthProviderWireEvent, { event: event }>["params"],
 ) => void;
+
+export type EventFrameAddRejected = {
+  event: "frame_add_rejected";
+  reason: AddFrameRejectedReason;
+};
+
+export type EventPrimaryButtonClicked = {
+  event: "primary_button_clicked";
+};
+
+export type SdkFrameEvent =
```

Comment:
these names are slightly confusing and differing definitions are confusing

could we call FrameEvent -> FrameServerEvent and only fire these on the serverside?

Reply:
I want the payloads to be the same. So I'm going to add a new name for backend events and frontend events with unions